constructor/main.py

```diff
@@ -197,6 +197,15 @@ def main_build(dir_path, output_dir='.', platform=cc_platform,
             )
         )
 
+    # Add --no-rc option to CONDA_EXE command so that existing
+    # .condarc files do not pollute the installation process.
+    if exe_name == "conda-standalone" and Version(exe_version) >= Version("24.9.0"):
+        info["_ignore_condarcs_arg"] = "--no-rc"
+    elif exe_name == "micromamba":
+        info["_ignore_condarcs_arg"] = "--no-rc"
+    else:
+        info["_ignore_condarcs_arg"] = ""
+
```

Comment:
That hardcoding of the names of binaries that constructor builds is a dealbreaker, let's find a different way

Reply:
Is your objection here that I use an if-else chain or that I use strings? The latter is the result of the `identify_conda_exe` function, so changing that will widen the scope of this PR.

Identifying whether the standalone function is `conda-standalone` is done inconsistently (see, e.g., here: https://github.com/conda/constructor/blob/49b80727bbdabce4d150e3fba4d03c953574ea1f/constructor/main.py#L95), but I wonder if a separate PR would be a better place to address the issue.

Reply:
Here is a PR that addresses the issue: https://github.com/conda/constructor/pull/864

That would get rid of string comparisons at least and removes inconsistencies.